core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.kafka.scaladsl
+
+import scala.concurrent.Future
+import akka.Done
+import akka.stream.scaladsl.Source
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import akka.kafka.internal.ConsumerStage
+import akka.NotUsed
+import akka.dispatch.ExecutionContexts
+import scala.concurrent.ExecutionContext
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.clients.consumer.OffsetAndMetadata
+import akka.kafka.internal.ConsumerStage.CommittableOffsetBatchImpl
+import akka.kafka.internal.PlainConsumerStage
+import akka.kafka.internal.CommittableConsumerStage
+
+object Consumer {
+
+  trait Message[K, V] {
+    def key: K
+    def value: V
+    def offset: PartitionOffset
+  }
+
+  final case class CommittableMessage[K, V](
+      key: K,
+      value: V,
+      committableOffset: CommittableOffset
+  ) extends Message[K, V] {
+
+    override def offset: PartitionOffset = committableOffset.offset
+  }
+
+  /**
+   * This interface might move into `akka.stream`
+   */
+  trait Committable {
+    def commit(): Future[Done]
+  }
+
+  trait CommittableOffset extends Committable {
+    def offset: PartitionOffset
+  }
+
+  final case class PartitionOffset(topic: String, partition: Int, offset: Long)
+
+  object CommittableOffsetBatch {
```

Comment:
I try to move batcher implementation from impl package to user level to allow users to implement batching in their ways. What do you think about this?

I am not sure about efficiency of commits on kafka client level. Right now (0.9.0.1) `commitAsync` just schedules commits to be performed in next poll session. But in 0.10.0 it can be changed (https://issues.apache.org/jira/browse/KAFKA-3412 and https://github.com/apache/kafka/pull/1108). Looks like that that in 0.10 commit batching will be more efficient.